packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1469,6 +1469,32 @@ export const UnknownContentCollectionError = {
 	name: 'UnknownContentCollectionError',
 	title: 'Unknown Content Collection Error.',
 } satisfies ErrorData;
+
+/**
+ * @docs
+ * @description
+ * Astro encountered an unexpected error when writing your content layer data to disk.
+ *
+ * If you can reliably cause this error to happen, we'd appreciate if you could [open an issue](https://astro.build/issues/)
+ */
+export const ContentLayerWriteError = {
```

Comment:
A similar error was added for injectType, I think we make it use teh same one

Reply:
Do you want a include any renaming/rewording in this PR, or will you do it in yours?

Reply:
I'll merge them post merge

Reply:
Done!